test/e2e/egress_test.go

```diff
@@ -664,6 +666,89 @@ func testEgressMigration(t *testing.T, data *TestData, triggerFunc, revertFunc f
 	checkIPNeighbor(fromNode)
 }
 
+func testEgressUpdateBandwidth(t *testing.T, data *TestData) {
+	skipIfNotIPv4Cluster(t)
+	skipIfHasWindowsNodes(t)
+	var (
+		bandwidth1 = &v1beta1.Bandwidth{
+			Rate:  "100M",
+			Burst: "200M",
+		}
+		bandwidth2 = &v1beta1.Bandwidth{
+			Rate:  "1G",
+			Burst: "2G",
+		}
+		transMap = map[string]int{
+			"100M": 100,
+			"200M": 200,
+			"1G":   1000,
+			"2G":   2000,
+		}
+	)
+
+	egressNode := controlPlaneNodeName()
+	egressNodeIP := controlPlaneNodeIPv4()
+
+	// Create another netns to fake an external network on the host network Pod.
+	fakeExternalName := "fake-external"
+	fakeExternalCmd := "iperf3 -s"
+	cmd, _ := getCommandInFakeExternalNetwork(fakeExternalCmd, 24, "1.1.1.1", "1.1.1.254")
+	if err := NewPodBuilder(fakeExternalName, data.testNamespace, toolboxImage).OnNode(egressNode).WithCommand([]string{"sh", "-c", cmd}).InHostNetwork().Privileged().Create(data); err != nil {
+		t.Fatalf("Failed to create client Pod: %v", err)
+	}
+	defer deletePodWrapper(t, data, data.testNamespace, fakeExternalName)
+	if err := data.podWaitForRunning(defaultTimeout, fakeExternalName, data.testNamespace); err != nil {
+		t.Fatalf("Error when waiting for Pod '%s' to be in the Running state", fakeExternalName)
+	}
+
+	clientPodName := "client-pod"
+	if err := NewPodBuilder(clientPodName, data.testNamespace, toolboxImage).OnNode(egressNode).Create(data); err != nil {
+		t.Fatalf("Error when creating the perftest client Pod: %v", err)
+	}
+	defer deletePodWrapper(t, data, data.testNamespace, clientPodName)
+	if err := data.podWaitForRunning(defaultTimeout, clientPodName, data.testNamespace); err != nil {
+		t.Fatalf("Error when waiting for the perftest client Pod: %v", err)
+	}
+
+	egress := data.createEgress(t, "egress-qos-", nil, map[string]string{"antrea-e2e": clientPodName}, "", egressNodeIP, bandwidth1)
+	_, err := data.waitForEgressRealized(egress)
+	if err != nil {
+		t.Fatalf("Error when waiting for Egress to be realized: %v", err)
+	}
+
+	// expectedBandwidth is Mbps
+	runIperf := func(cmd []string, expectedBandwidth int) {
+		stdout, _, err := data.RunCommandFromPod(data.testNamespace, clientPodName, "toolbox", cmd)
+		if err != nil {
+			t.Fatalf("Error when running iperf3 client: %v", err)
+		}
+		stdout = strings.TrimSpace(stdout)
+		actualBandwidth, _ := strconv.ParseFloat(strings.TrimSpace(stdout), 64)
+		t.Logf("Actual bandwidth: %v Mbits/sec", actualBandwidth)
+		// Allow a certain deviation.
+		assert.InEpsilon(t, actualBandwidth, expectedBandwidth, 0.1)
+	}
+
+	runIperf([]string{"bash", "-c", "iperf3 -c 1.1.1.1 -f m -O 1|grep sender|awk '{print $7}'"}, transMap[bandwidth1.Rate]+transMap[bandwidth1.Burst])
```

Comment:
If ignoring the 1st second, could it reach the desired speed steadily?

Reply:
Changed to only capture the first second to test the burst. And skip the first second to test the rate.